document/core/exec/instructions.rst

```diff
@@ -2680,14 +2680,40 @@ Control Instructions
 :math:`\RETHROW~l`
 ..................
 
+1. Assert: due to :ref:`validation <valid-rethrow>`, the stack contains at least :math:`l+1` labels.
+
+2. Let :math:`L` be the :math:`l`-th label appearing on the stack, starting from the top and counting from zero.
+
+3. Assert: due to :ref:`validation <valid-rethrow>`, :math:`L` is a catch label, i.e., a label of the form :math:`(\LCATCH~[t^\ast])`.
+
+4. Repeat :math:`l+1` times:
```

Comment:
I suggest doing it only l times in the loop, so that you haven't skipped over the caught clause already.

Reply:
I removed the loop entirely, trying to avoid popping and pushing back the entire block context.